### PR TITLE
WT-12630 Fix compilation Clang 17

### PR DESCRIPTION
**Pull request by korteland**
Tested on my personal machine running Clang 17.0.6.

### CODE REVIEW

File: cmake/strict/strict_flags_helpers.cmake [hunk 1]
```diff
@@ -121,6 +124,10 @@ function(get_clang_base_flags flags)
     list(APPEND clang_flags "-Wno-packed")
     list(APPEND clang_flags "-Wno-padded")
     list(APPEND clang_flags "-Wno-reserved-id-macro")
+    # FIXME-WT-11788: Remove the following flag once the violation of the standard C11 7.1.3 has
+    # been addressed.
+    list(APPEND clang_flags "-Wno-reserved-identifier")
+    list(APPEND clang_flags "-Wno-unsafe-buffer-usage")
```
Comment (etienneptl) [inline cmake/strict/strict_flags_helpers.cmake:130]: Is that a flag we want to keep or is it a short term workaround as well?

**Comment by marcbutler** [comment]
It's a really good question. Do we upgrade the toolchains for retrospective
builds? attn @Luke Chen ***@***.***>

File: cmake/strict/strict_flags_helpers.cmake (hunk 1 above)
Comment (korteland) [reply cmake/strict/strict_flags_helpers.cmake:130]: "Keep for the foreseeable future" is my current thinking - this flag is pretty broken right now (https://discourse.llvm.org/t/rfc-c-buffer-hardening/65734/85).

**Comment by korteland** [comment]
PR testing failure is unrelated to this change.

**Comment by marcbutler** [comment]
> PR testing failure is unrelated to this change.

Thanks for the heads up.

File: cmake/strict/strict_flags_helpers.cmake (hunk 1 above)
Comment (marcbutler) [reply cmake/strict/strict_flags_helpers.cmake:130]: The reserved identifier flag is fine as well, the identifiers in v7 are in no way less reserved. 😀

File: test/evergreen/configure_combinations.sh [hunk 2]
```diff
@@ -37,7 +37,9 @@ echo `pwd`
 
 curdir=`pwd`
 
-flags="-DCMAKE_TOOLCHAIN_FILE=$curdir/cmake/toolchains/gcc.cmake -DCMAKE_C_FLAGS=\"-Werror -Wall -Wextra -Waddress -Waggregate-return -Wbad-function-cast -Wcast-align -Wdeclaration-after-statement -Wformat-security -Wformat-nonliteral -Wformat=2 -Wmissing-declarations -Wmissing-field-initializers -Wmissing-prototypes -Wnested-externs -Wno-unused-parameter -Wpointer-arith -Wredundant-decls -Wshadow -Wundef -Wunused -Wwrite-strings -O -fno-strict-aliasing -Wuninitialized\"
+# FIXME-WT-11788: Remove the -Wno-reserved-identifier flag once the violation of the standard
+# C11 7.1.3 has been addressed.
+flags="-DCMAKE_TOOLCHAIN_FILE=$curdir/cmake/toolchains/gcc.cmake -DCMAKE_C_FLAGS=\"-Werror -Wall -Wextra -Waddress -Waggregate-return -Wbad-function-cast -Wcast-align -Wdeclaration-after-statement -Wformat-security -Wformat-nonliteral -Wformat=2 -Wmissing-declarations -Wmissing-field-initializers -Wmissing-prototypes -Wnested-externs -Wno-reserved-identifier -Wno-unused-parameter -Wpointer-arith -Wredundant-decls -Wshadow -Wundef -Wunused -Wwrite-strings -O -fno-strict-aliasing -Wuninitialized\"
```
Comment (marcbutler) [inline test/evergreen/configure_combinations.sh:42]: Something to consider: afaict in CMake we can pull in external file containing the list of flags, potentially singular source for cmake and this script.
https://cmake.org/cmake/help/v3.11/command/file.html

Comment (korteland) [reply test/evergreen/configure_combinations.sh:42]: I don't mind the idea but I think it's out of scope for this PR. I can make a ticket if you like?

**Review by etienneptl — APPROVED** [review]

File: test/evergreen/configure_combinations.sh (hunk 2 above)
Comment (korteland) [reply test/evergreen/configure_combinations.sh:42]: Made WT-12965.

Comment (marcbutler) [reply test/evergreen/configure_combinations.sh:42]: thank you

**Review by marcbutler — APPROVED** [review]
LGTM

**Pull request merged**